crates/api_common/src/request.rs

```diff
@@ -200,50 +205,59 @@ pub async fn fetch_site_data(
       // Warning, this may ignore SSL errors
       let metadata_option = fetch_site_metadata(client, url).await.ok();
       if !include_image {
-        return (metadata_option, None);
+        (metadata_option, None)
+      } else {
+        let thumbnail_url =
+          match fetch_pictrs_hash(client, &metadata_option, settings, url).await {
+            Ok(pictrs_hash) => Url::parse(&format!(
+              "{}/pictrs/image/{}",
+              settings.get_protocol_and_hostname(),
+              pictrs_hash
+            ))
+            .map(Into::into)
```

Comment:
Hrm... this already returns a LemmyError... not sure why this is necessary. And you shouldn't need to check the type of error either, doing a `.ok()` should replace any error with `None`

Reply:
`Url::parse` returns a `ParseError`, not a `LemmyError`.